src/operator/controllers/intents/reconciler.go

```diff
@@ -299,6 +311,17 @@ func (r *Reconciler) handleDBUserCreation(
 	return nil
 }
 
+func (r *Reconciler) isNamespaceTerminating(ctx context.Context, namespace string) (bool, error) {
```

Comment:
Is this true? Doesn't it delete users?
It's also not true for other types of credentials, so the reconciler should refer to user password. I know it's just the filename but if somebody were to add logic for other types to this generic intents reconciler, it might be a mistake (other types do cleanup on termination)

Reply:
Users are deleted by the intents operator (you must clean permissions before dropping users in PG, so it fits better there). There's no cleanup being made here.
I'm not sure what's the action I should following this comment. Rename the file ? Add a warning ? That's why I left a comment initially.